primus.js

```diff
@@ -823,51 +816,26 @@ Primus.prototype._write = function write(data) {
 };
 
 /**
- * Send a new heartbeat over the connection to ensure that we're still
- * connected and our internet connection didn't drop. We cannot use server side
- * heartbeats for this unfortunately.
+ * Set a timer that, upon expiration, closes the client.
  *
  * @returns {Primus}
  * @api private
  */
 Primus.prototype.heartbeat = function heartbeat() {
-  var primus = this;
-
-  if (!primus.options.ping) return primus;
-
-  /**
-   * Exterminate the connection as we've timed out.
-   *
-   * @api private
-   */
-  function pong() {
-    primus.timers.clear('pong');
+  if (!this.options.ping) return this;
 
+  this.timers.clear('ping');
+  this.timers.setTimeout('ping', function ping() {
```

Comment:
I think this timer is still needed. Without this the client will never know if the connection has been closed abruptly (no FIN patchet etc.).